connectors/sources/graphql.py

```diff
@@ -0,0 +1,333 @@
+        try:
+            if self.http_method == "get":
+                return await self.get(graphql_query=graphql_query)
+            else:
+                return await self.post(graphql_query=graphql_query)
+        except ClientResponseError as exception:
+            if exception.status == 401:
+                msg = "Provided credentials or token do not have the necessary permissions to perform the request."
+                raise UnauthorizedException(msg) from exception
+            else:
+                raise
+        except Exception:
+            raise
+
+    async def get(self, graphql_query):
+        params = {"query": graphql_query}
+        async with self._get_session.get(url=self.url, params=params) as response:
+            json_response = await response.json()
+            if not json_response.get("errors"):
+                data = json_response.get("data", {})
+                return data
+            msg = f"Error while executing query. Exception: {json_response['errors']}"
+            raise Exception(msg)
+
+    async def post(self, graphql_query):
+        """Invoke GraphQL request to fetch response.
+
+        Raises:
+            UnauthorizedException: Unauthorized exception.
+            ClientResponseError: ClientResponseError exception.
+            exception: An instance of an exception class.
+
+        Yields:
+            dictionary: Client response
+        """
+        query_data = {
+            "query": graphql_query,
+            "variables": self.variables,
+        }
+        async with self._get_session.post(url=self.url, json=query_data) as response:
+            json_response = await response.json()
+            if not json_response.get("errors"):
+                data = json_response.get("data", {})
+                return data
+            msg = f"Error while executing query. Exception: {json_response['errors']}"
+            raise Exception(msg)
+
+    async def close(self):
+        self._sleeps.cancel()
+        await self._get_session.close()
+        del self._get_session
+
+    async def ping(self):
+        await self.make_request(graphql_query=PING_QUERY)
+
+
+class GraphQLDataSource(BaseDataSource):
+    """GraphQL"""
+
+    name = "GraphQL"
+    service_type = "graphql"
+
+    def __init__(self, configuration):
+        """Setup the connection to the GraphQL instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.graphql_client = GraphQLClient(configuration=configuration)
+
+    def _set_internal_logger(self):
+        self.graphql_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GraphQL.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "http_endpoint": {
```

Comment:
Perhaps we ca add some validation, e.g. regex to match that the endpoint is structured correctly. 

Are we only supporting `http` protocol, or is it possible to connect to graphql over `https`?

Reply:
Sure added validation for endpoints and yes connector supports both `http` and `https`.